views.py

```diff
@@ -425,3 +444,160 @@ def edit_supporter_band(request, supporter_id=None):
     return render(request, template, context)
 
 
+def recommend_us(request):
+    """
+    A view for authors, editors, readers, and others
+    to generate an email for recommending us to their librarian.
+    Depends on the subsequent views via HTMX.
+    """
+    template = hourglass('custom/recommend-us.html')
+    context = {
+        'step': '1',
+    }
+    return render(request, template, context)
+
+
+@require_POST
+def recommend_us_see_matching_supporters(request):
+    query = request.POST.get('supporter_query', '')
+    if query:
+        supporters = supporter_models.Supporter.objects.filter(name__icontains=query)
+    else:
+        supporters = supporter_models.Supporter.objects.none()
+    template = hourglass('custom/recommend-us-see-matching-supporters.html')
+    context = {
+        'supporters': supporters,
+        'step': '2',
+        'query': query,
+    }
+    return render(request, template, context)
+
+
+@require_POST
+def recommend_us_choose_role(request):
+    template = hourglass('custom/recommend-us-choose-role.html')
+    context = {
+        'step': '3',
+    }
+    return render(request, template, context)
+
+
+@require_POST
+def recommend_us_search_article(request):
+    context = {
+        'step': '4',
+    }
+    if 'author' in request.POST:
+        context['role'] = 'author'
+    elif 'reader' in request.POST:
+        context['role'] = 'reader'
+    template = hourglass('custom/recommend-us-search-article.html')
+    return render(request, template, context)
+
+
+@require_POST
+def recommend_us_search_journal(request):
+    context = {
+        'role': 'editor',
+        'step': '4',
+    }
+    template = hourglass('custom/recommend-us-search-journal.html')
+    return render(request, template, context)
+
+
+@require_POST
+def recommend_us_choose_article(request):
+    query = request.POST.get('article_query', '')
+    if query:
+        articles = request.press.published_articles.filter(title__icontains=query)
```

Comment:
It is probably worth using `Article.objects.search`, with just the article title filter so that we leverage our full text search:

```
    def search(self, search_term, search_filters, sort=None, site=None):
```

icontains doesn't work particularly well for searching phrases.